src/bindings/python/tests/test_graph/test_custom_op.py

```diff
@@ -197,3 +199,89 @@ def test_custom_op():
     input_tensor = Tensor(input_data)
     results = request.infer({"data": input_tensor})
     assert np.allclose(results[list(results)[0]], expected_output, 1e-4, 1e-4)
+
+
+class CustomSimpleOp(Op):
+    def __init__(self, inputs=None):
+        super().__init__(self, inputs)
+
+    def validate_and_infer_types(self):
+        self.set_output_type(0, self.get_input_element_type(0), self.get_input_partial_shape(0))
+
+
+class CustomSimpleOpWithAttribute(Op):
+    class_type_info = DiscreteTypeInfo("CustomSimpleOpWithAttribute", "extension")
+
+    def __init__(self, inputs=None, **attrs):
+        super().__init__(self, inputs)
+        self._attrs = attrs
+
+    def validate_and_infer_types(self):
+        self.set_output_type(0, self.get_input_element_type(0), self.get_input_partial_shape(0))
+
+    @staticmethod
+    def get_type_info():
+        return CustomSimpleOpWithAttribute.class_type_info
+
+    def visit_attributes(self, visitor):
+        if self._attrs:
+            visitor.on_attributes(self._attrs)
+        return True
+
+
+def test_op_extension(prepared_paths):
+    input_shape = [2, 1]
+
+    core = Core()
+    core.add_extension(CustomSimpleOp)
+    core.add_extension(OpExtension(CustomSimpleOpWithAttribute))
+    core.add_extension(OpExtension(CustomAdd))
+
+    param1 = ops.parameter(Shape(input_shape), dtype=np.float32, name="data1")
+    param2 = ops.parameter(Shape(input_shape), dtype=np.float32, name="data2")
+    custom_simple = CustomSimpleOp([param1, param2])
```

Comment:
Will `CustomSimpleOp(param1, param2)` way be available as a more native illustration? I would illustrate positional arguments instead of a list as the default design choice for users who want to implement their ops. @akuporos, do you see troubles with that approach?

Reply:
I added it